pkg/asset/installconfig/aws/validation.go

```diff
@@ -329,12 +330,23 @@ func validateEndpointAccessibility(endpointURL string) error {
 	if port == "" {
 		port = "https"
 	}
-	conn, err := proxy.Dial(context.Background(), "tcp", net.JoinHostPort(URL.Hostname(), port))
-	if err != nil {
-		return err
+	err = dialURL(context.Background(), net.JoinHostPort(URL.Hostname(), port))
+	return err
+}
+
+func dialURL(ctx context.Context, URL string) error {
+	// DefaultClient picks the http and https proxy environment variables.
+	_, err := http.Get(URL)
+	if err == nil {
+		return nil
+	}
+	// Proxy Dial picks up the ALL_PROXY environment variable and handles sock5 proxy dialing to URL.
+	conn, err := proxy.Dial(context.Background(), "tcp", URL)
+	if err == nil {
```

Comment:
Probably better to check if the conn != nil to close the connection instead of the value of err. Something like this:
    if conn != nil {
        conn.Close()
    }
    return err